src/core/core/task.cpp

```diff
@@ -366,18 +367,18 @@ void task::enqueue()
 void task::enqueue(task_worker_pool* pool)
 {
     this->add_ref(); // released in exec_internal (even when cancelled)
-    
+
+    if (spec().type == TASK_TYPE_COMPUTE)
+    {
```

Comment:
this causes the timer tasks be enqueued twice in logs, but i think it is ok as we now have  more information